docs/foreign-keys.md

```diff
@@ -0,0 +1,330 @@
+
+```json
+{
+    "title": "RecordType of the related Record",
+    "type": "string",
+    "format": "select",
+    "enumSource": [{
+        "source": [
+            {
+                "title": "<RecordType.title>",
+                "value": "<RecordType.uuid>"
+            },
+            ...More RecordTypes from the API
+        ]
+    }]
+}
+```
+
+Then, when saving out the final schema in `schemas-service.js`, the
+RecordType UUID that the user had selected in the `recordType` field would be used to define
+the available Record values for reference. Again, the schema would make use of the
+`$ref` attribute to load valid Records dynamically through a Grout API endpoint
+during validation:
+
+```javascript
+{
+    "<propertyName>": {
+        "allOf": [{
+            "$ref": "https://<grout-server-hostname>/api/record-types/<UUID>/referents.json"
+        }]
+    }
+}
+```
+
+In this case, the `/api/record-types/<UUID>/referents.json` endpoint would
+resolve to a type definition listing available Records: 
+
+```
+{
+    "type": "string",
+    "format": "select",
+    "enumSource": [{
+        "source": [
+            {
+                "title": "<Digest for the Record>"
+                "value": "<Record UUID>"
+            },
+            ...More records from the API
+        ]
+    }]
+}
+```
+
+Some import considerations for this pattern:
+
+- Unlike in [Derek's initial comment suggesting this
+  pattern](https://github.com/azavea/grout-2018-fellowship/pull/34/files#r203071181),
+  there would have to be changes to the schema editor (specifically
+  `schemas-service.js`) in order to support the new fields. This is necessary
+  because there has to be point in the process that takes the RecordType selected by the user
+  and translates it into a `$ref` endpoint for validating Records. However,
+  this approach still has the benefit of leveraging `$ref` to push all of
+  the validation burden onto the JSON Schema, without having to write custom
+  validators.
+
+- It's not totally clear to us yet whether validation with external `$refs`
+  will work in JSONSchema draft 4, which is the latest version supported by
+  JSON Editor. See [this thread](https://github.com/azavea/grout-2018-fellowship/pull/34/files#r203071181)
+  for some discussion. Reference validation needs to be confirmed before
+  moving ahead with this pattern.
+
+- Assuming Records are append-only, meaning they cannot be deleted from the
+  database through the Grout API, this pattern should always validate properly
+  as long as the `/api/record-types/<UUID>/referents.json` endpoint returns a
+  list of all active Records as well as all archived Records. However, returning
+  all Records for a RecordType regardless of whether or not they're active could
+  potentially be confusing for an end user doing data collection work, both
+  because it might be hard to tell which Records are active and because the
+  endpoint may return overwhelmingly large lists of Records. Records
```

Comment:
Another potential issue is that the `referents.json` endpoint could end up timing out or blowing up the memory of the JSON-Schema validator if the number of records is very large.

This also might require `referents.json` to be public, which could be problematic under some situations. Ordinarily I wouldn't think that a list of Record UUIDs would be an issue but depending on the type of information being stored, just the count of active records could potentially be sensitive.

Reply:
Great points! I added two notes to this section based on your comment.